scripts/adb_devices_abis.sh

```diff
@@ -0,0 +1,40 @@
+#!/bin/bash
+
+set -e
+typeset -a abis
+SCRIPT_NAME=$(basename $0)
+
+# make sure server is running, otherwise sdb will show devices offline
+adb start-server
+
+adb_devices=$(adb devices | tail -n+2)
+
+while IFS= read line; do
+  read -a arr <<< "${line}"
+
+  device_name="${arr[0]}"
+  device_status=${arr[1]}
+
+  echo "${SCRIPT_NAME} DEBUG: device_name=${device_name}" >&2
+  echo "${SCRIPT_NAME} DEBUG: device_status=${device_status}" >&2
```

Comment:
If you're repeating this a lot, just use a function:
```sh
function debug() {
  echo "${SCRIPT_NAME} DEBUG: ${@}" >&2
}
```
```suggestion
  debug "device_name=${device_name}"
  debug "device_status=${device_status}"
```